cmake/linker_script/arm/linker.cmake

```diff
@@ -83,6 +83,8 @@ zephyr_linker_section_configure(SECTION .text INPUT ".glue_7t")
 zephyr_linker_section_configure(SECTION .text INPUT ".glue_7")
 zephyr_linker_section_configure(SECTION .text INPUT ".vfp11_veneer")
 zephyr_linker_section_configure(SECTION .text INPUT ".v4_bx")
+zephyr_linker_section_configure(SECTION .text INPUT ".shared_irq")
```

Comment:
.text section is before .device, in final link, it will change .device section address.